### ISSUE TITLE
Incorrect namespace with anonymous top level

### ISSUE THREAD

**Issue by cbandy** [go-playground/validator#266]
### Package version eg. v8, v9: 

`gopkg.in/go-playground/validator.v9`

### Issue, Question or Enhancement:

When the top-level struct is anonymous, diving into a slice of structs returns FieldError with an incorrect namespace.

I started with an anonymous struct and a `json` TagNameFunc. Things went well:

```go
var input struct {
  Children []string `json:"children" validate:"required,gt=0,dive,required"`
}
input.Children = []string{"ok", ""}

validate.Struct(input)

/*
  FieldError{
    Namespace(): `children[1]`
    Field():     `children[1]`
	
    StructNamespace(): `Children[1]`
    StructField():     `Children[1]`
  }
*/
```

When my needs grew beyond a slice of string, I refactored to a slice of struct. The namespace returned no longer has an index nor the top-level field name:

```go
type child struct {
  Name string `json:"name" validate:"required"`
}
var input struct {
  Children []child `json:"children" validate:"required,gt=0,dive"`
}
input.Children = []child{{"ok"}, {""}}

validate.Struct(input)

/*
  FieldError{
    Namespace(): `child.name`       // I expected `children[1].name`
    Field():     `name`
	
    StructNamespace(): `child.Name` // I expected `Children[1].Name`
    StructField():     `Name`
  }
*/
```

I found that using a named top-level type can work around this, but now I also have to trim a level from the returned namespace:

```go
type child struct {
  Name string `json:"name" validate:"required"`
}
type parent struct {
  Children []child `json:"children" validate:"required,gt=0,dive"`
}

var input parent
input.Children = []child{{"ok"}, {""}}

validate.Struct(input)

/*
  FieldError{
    Namespace(): `parent.children[1].name`
    Field():     `name`
	
    StructNamespace(): `parent.Children[1].Name`
    StructField():     `Name`
  }
*/
```

**Comment by joeybloggs** [comment]
Hello @cbandy 

I haven't seen that one before. 

I immediately suspect that it because the struct used in your array is unexposed(lowercase name) which will limit the I formation the validator package can get from the struct, but I will look into this and verify exactly what's happening.

If you wanted to try with an exposed struct in the mean time that would sure speed things along :)

Thanks

**Comment by joeybloggs** [comment]
Hey @cbandy 

I've corrected the issue in [Release 9.3.2](https://github.com/go-playground/validator/releases/tag/v9.3.2) 

please let me know if it solves the issue for you.

**Comment by cbandy** [comment]
Resolved! Thank you.
